Src/IronPython/Modules/Builtin.cs

```diff
@@ -925,18 +934,23 @@ public class mapobject : IEnumerable {
         }
 
         public static object max(CodeContext/*!*/ context, object x, object y, [ParamDictionary] IDictionary<object, object> dict) {
-            object method = GetMaxKwArg(dict);
+            var kwargTuple = GetMaxKwArg(dict);
+            object method = kwargTuple.Item1;
+
             PythonContext pc = context.LanguageContext;
             return pc.GreaterThan(PythonCalls.Call(context, method, x), PythonCalls.Call(context, method, y)) ? x : y;
         }
 
         public static object max(CodeContext/*!*/ context, [ParamDictionary]IDictionary<object, object> dict, params object[] args) {
+            var kwargTuple = GetMaxKwArg(dict);
+            object method = kwargTuple.Item1;
+            object def = kwargTuple.Item2;
```

Comment:
default is not used here?

Reply:
Seems like with python 3 i get the error below. I will try to trigger it another way. If there are some tests cases I would be glad to try to fix this section.
TypeError: Cannot specify a default for max() with multiple positional arguments
>>> max([], True, True, key=lambda x: x, default=1)

Reply:
I found some test cases in the main repo. i'll take a look